jax/lax/lax.py

```diff
@@ -2705,19 +2722,31 @@ def _dynamic_slice_shape_rule(operand, start_indices, slice_sizes,
     raise TypeError(msg.format(slice_sizes))
   return tuple(slice_sizes)
 
-def _dynamic_slice_translation_rule(c, operand, start_indices, slice_sizes,
-                                    operand_shape):
+def _dynamic_slice_translation_rule(c, operand, *start_indices,
+                                    slice_sizes=None, operand_shape=None):
   return c.DynamicSlice(operand, start_indices, slice_sizes)
 
-def _dynamic_slice_jvp_rule(g, operand, start_indices, slice_sizes,
-                            operand_shape):
-  return dynamic_slice(g, start_indices, slice_sizes)
+def _dynamic_slice_jvp(primals, tangents, slice_sizes, operand_shape):
+  tangent_out = ad_util.zero
+  if tangents[0] is not ad_util.zero:
+    tangent_out = dynamic_slice(tangents[0], primals[1:], slice_sizes)
+  return dynamic_slice(primals[0], primals[1:], slice_sizes), tangent_out
 
-def _dynamic_slice_transpose_rule(t, operand, start_indices, slice_sizes,
-                                  operand_shape):
+def _dynamic_slice_transpose_rule(t, operand, *start_indices, slice_sizes=None,
+                                  operand_shape=None):
   assert operand is None
   zeros = full(operand_shape, tie_in(t, _zero(t)))
-  return [dynamic_update_slice(zeros, t, start_indices), ad_util.zero]
+  return ([dynamic_update_slice(zeros, t, start_indices)] +
+          [ad_util.zero] * len(start_indices))
```

Comment:
I think no element of `start_indices` can be None here, so:

1. we could `assert all(s is not None for s in start_indices)` (since it's required by the call to `dynamic_update_slice`), and
2. instead of `[ad_util.zero] * len(start_indices)` we should return in its place `[None] * len(start_indices)`.

Reply:
Done.